### ISSUE TITLE
Change cursor color?

### ISSUE THREAD

**Issue by prurph** [catppuccin/jupyterlab#9]
I rarely use Jupyter so this is likely a dumb question: but is this theme supposed to alter the cursor color? It's staying the default lime green for me which is hard to see with almost any theme.

Is it possible to set or override this? I took a very quick look at the CSS in dev tools but didn't immediately see what class was responsible for the cursor color. Thanks!

![CleanShot 2024-06-15 at 13 44 36@2x](https://github.com/catppuccin/jupyterlab/assets/6109531/78929d41-a63c-43a2-91af-bddfc7844e96)

**Comment by gabrielmagno** [comment]
Hello there @prurph !

It is indeed possible to set the cursor color in Jupyter Lab theme, we just have to set the CSS variable `--jp-editor-cursor-color`.

Our Catppuccin theme is in fact already setting it as this:
```
--jp-editor-cursor-color: var(--ctp-plt-rosewater);
```

So the cursor will be in the color "rosewater" of the corresponding Catppuccin flavor that is being used.
In case of the Latte flavor, it is `#dc8a78`.

I've just tested the Latte flavor (both in Firefox and Chrome) and it seems to working:
![image](https://github.com/catppuccin/jupyterlab/assets/244890/d0b69304-29f4-4b50-9cb8-fe66b95d0d76)

Not sure why your cursor is green. Also, don't know why it is as a "block cursor", and not a "line cursor".
Is it a JupyterLab setting? 

Which browser are you using? Could you test in other browsers?

**Comment by prurph** [comment]
Hi there! The issue seems to be that I'm using vim mode (Settings → Enable Vim Mode). The cursor is the green block in normal mode. In insert mode the line cursor is colored as it should be. I had tried a few ways of overriding it with custom css but your snippet inspired me to dig in a little. The correct variable is `--jp-vim-cursor-color` [link](https://github.com/jupyterlab-contrib/jupyterlab-vim/blob/main/style/base.css#L21C3-L21C25), but overriding this in custom css requires attaching `!important` (it looks like the vim styles are injected _after_ custom user styles).

The final fix is:

```css
/* ~/.jupyter/custom/custom.css */
:root {
  --jp-vim-cursor-color: var(--ctp-plt-rosewater) !important;
}
```

and voila 💅

<img width="194" alt="image" src="https://github.com/catppuccin/jupyterlab/assets/6109531/17fcc8d0-9041-421b-be0b-f28953fb503d">

I would be happy to prep a small PR to add this to to the catppuccin plugin if you think this is appropriate, but I couldn't figure out where these extensions are writing their data to disk so I could do a quick sanity check of whether the change in catppuccin would even have an effect; I suspect the variable might just get overridden by the vim plugin (because I needed to add `!important` or it wouldn't work). I didn't see anything in any of the directories `jupyter --paths` that had actual data in it to test.

If you know offhand how I could quickly test this locally by manually editing some plugin files I'd be happy to do this; otherwise the custom.css is an easy solution for other users. Thanks for the help and the spirited theme! 😸

**Comment by gabrielmagno** [comment]
Hey @prurph. Great discovery! 

So the main issue is that the style comes from an external plugin, that sets its own cursor, with a [hard-coded color](https://github.com/jupyterlab-contrib/jupyterlab-vim/blob/b0a90a93b54d55df617f462c209c9fbe94d835d4/style/base.css#L21).

This probably affects any Jupyter Lab theme, not only catppuccin. Ideally this should be a fix/feature of the plugin, either to allow the user to set its own color, or to inherit the color from the jupyter css styles instead of associating a hard coded fixed color.

That being said, I would be happy to try making a [gambiarra](https://www.urbandictionary.com/define.php?term=Gambiarra) to make catppuccin theme compatible with the vim plugin. I'm afraid that it might not be possible tough. 

I tried setting the `jp-vim-cursor-color` (with !important flag) like you said, inside the `variable.css` file of the catppuccin theme, but it seems to have no effect. I don't have much experience with advanced CSS composition, so maybe you will be more effective than I am!

You can test and modify the catppuccin theme locally by doing this:
1. Clone the repository: `git clone https://github.com/catppuccin/jupyterlab.git`;
2. Load the environment where your Jupyter Lab will run;
3. Change directory to be on the root of the repository;
4. Install the theme from local folder: `pip install -e .`;
5. Open Jupyter Lab: `jupyter lab`;

So you could keep editing the repository and theme files, reinstalling and reopening jupyter lab.

Please let me know if you can figure out a way to make it work 😺

**Comment by prurph** [comment]
@gabrielmagno obrigado pelas direções! I opened https://github.com/catppuccin/jupyterlab/pull/10, which correctly changes the block cursor with the vim extension on my machine. Not sure if you installed or have installed the vim extension in your tests, but if not then there will be no visible effect: this only causes the block cursor in vim insert mode to correctly match the color of the regular line cursor.

If you think it's weird to have another extension's variable set by catppuccin I totally get it; perhaps in that case we could abandon the PR and just add a note to the readme describing how vim users could create a custom.css file to override it themselves. Thanks!

**Comment by gabrielmagno** [comment]
Solved by #10
